index.js

```diff
@@ -54,17 +52,34 @@ module.exports = (password) => {
     ];
   }
 
+  if (new RegExp(`^${disallowed}`).test(password)) {
+    passwordContains = [
+      ...passwordContains,
+      {
+        message: "disallowed",
+      },
+    ];
+  }
+
   const strongRegex = new RegExp(
+    `^((${lowerCaseRegex}${upperCaseRegex}${numericRegex})|(${lowerCaseRegex}${upperCaseRegex}${symbolsRegex})|(${lowerCaseRegex}${numericRegex}${symbolsRegex})|(${upperCaseRegex}${numericRegex}${symbolsRegex}))(?=.{8,})`
+  );
+  const extraStrongRegex = new RegExp(
     `^${lowerCaseRegex}${upperCaseRegex}${numericRegex}${symbolsRegex}(?=.{8,})`
   );
   const mediumRegex = new RegExp(
     `^((${lowerCaseRegex}${upperCaseRegex})|(${lowerCaseRegex}${numericRegex})|(${upperCaseRegex}${numericRegex})|(${upperCaseRegex}${symbolsRegex})|(${lowerCaseRegex}${symbolsRegex})|(${numericRegex}${symbolsRegex}))(?=.{6,})`
   );
 
-  if (strongRegex.test(password)) {
+  if (extraStrongRegex.test(password)) {
+    strength = {
+      id: 3,
+      value: "Extra Strong",
+    };
+  } else if (strongRegex.test(password)) {
     strength = {
       id: 2,
-      value: "Strong",
+      value: "Strong - Can now submit",
```

Comment:
Is "Can now submit" necessary?

Reply:
No, I just put it in to provide confirmation to the user that a strong password is good enough for submission. It's not really necessary though.

Reply:
I would remove this then

Reply:
Removed additional text.